ecosystem/typescript/sdk_v2/src/internal/transaction.ts

```diff
@@ -32,3 +34,167 @@ export async function getGasPriceEstimation(args: { aptosConfig: AptosConfig })
   });
   return data;
 }
+
+export async function getTransactionByVersion(args: {
+  aptosConfig: AptosConfig;
+  txnVersion: AnyNumber;
+}): Promise<TransactionResponse> {
+  const { aptosConfig, txnVersion } = args;
+  const { data } = await getAptosFullNode<{}, TransactionResponse>({
+    aptosConfig,
+    originMethod: "getTransactionByVersion",
+    path: `transactions/by_version/${txnVersion}`,
+  });
+  return data;
+}
+
+export async function getTransactionByHash(args: {
+  aptosConfig: AptosConfig;
+  txnHash: string;
+}): Promise<TransactionResponse> {
+  const { aptosConfig, txnHash } = args;
+  const { data } = await getAptosFullNode<{}, TransactionResponse>({
+    aptosConfig,
+    path: `transactions/by_hash/${txnHash}`,
+    originMethod: "getTransactionByHash",
+  });
+  return data;
+}
+
+export async function isTransactionPending(args: { aptosConfig: AptosConfig; txnHash: string }): Promise<boolean> {
+  const { aptosConfig, txnHash } = args;
+  try {
+    const transaction = await getTransactionByHash({ aptosConfig, txnHash });
+    return transaction.type === TransactionType.Pending;
+  } catch (e: any) {
+    if (e?.status === 404) {
+      return true;
+    }
+    throw e;
+  }
+}
+
+/**
+ * Wait for a transaction to move past pending state.
+ *
+ * There are 4 possible outcomes:
+ * 1. Transaction is processed and successfully committed to the blockchain.
+ * 2. Transaction is rejected for some reason, and is therefore not committed
+ *    to the blockchain.
+ * 3. Transaction is committed but execution failed, meaning no changes were
+ *    written to the blockchain state.
+ * 4. Transaction is not processed within the specified timeout.
```

Comment:
```suggestion
 * 4. Transaction does not move past the pending state within `extraArgs.timeoutSecs` seconds.
```

Reply:
ooh I just realized we have the documentation copied in two places...hmm, we should probably link one to the other to avoid duplication/maintaining both

Maybe just explain in the other that it's just a wrapper for this function?

Reply:
I would leave the documentation in the other function actually, that would be the function developers interact with so better to have the explanation there